src/cppmangle.c

```diff
@@ -164,7 +164,8 @@ class CppMangleVisitor : public Visitor
                         else
                         {
                             dinteger_t val = e->toInteger();
-                            if (val < 0)
+                            // TODO: try and do this without knowing what dinteger_t type is
+                            if ((long)val < 0)
```

Comment:
Something like this?

```d
struct S(int x)
{
    int member;
};

extern(C++)
void func(S!(-1) s);

void main()
{
    func(S!(-1)(0));
}
```
```cpp

template<int x>
struct S
{
    int member;
};

void func(S<-1> s) {}
```

And add it to `runnable/cppa`.  If the mangling doesn't match it won't link successfully.